src/agent/Core/ApplicationPool/Group/SessionManagement.cpp

```diff
@@ -67,42 +67,57 @@ Group::route(const Options &options) const {
 		if (options.stickySessionId == 0) {
 			if (OXT_LIKELY(useNewRouting())) {
 				process = findBestProcess(enabledProcesses);
+				if (process != nullptr) {
+					assert(process->canBeRoutedTo());
+					return RouteResult(process);
+				} else {
+					return RouteResult(NULL, true);
+				}
             } else {
 				process = findEnabledProcessWithLowestBusyness();
-			}
-			if (process != nullptr) {
-				assert(process->canBeRoutedTo());
-				return RouteResult(process);
-			} else {
-				return RouteResult(NULL, true);
+				if (process && process->canBeRoutedTo()) {
+					return RouteResult(process);
+				} else {
+					return RouteResult(NULL, true);
```

Comment:
`process` should never be non-NULL here. But since we're being defensive, let's also change the `true` to `process == nullptr`.